pkg/api/validation/validation.go

```diff
@@ -2426,7 +2405,7 @@ func ValidatePreferredSchedulingTerms(terms []api.PreferredSchedulingTerm, fldPa
 // validatePodAffinityTerm tests that the specified podAffinityTerm fields have valid data
 func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopologyKey bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if !utilfeature.DefaultFeatureGate.Enabled(features.AffinityInAnnotations) && len(podAffinityTerm.TopologyKey) == 0 {
+	if len(podAffinityTerm.TopologyKey) == 0 {
```

Comment:
Wha tests? I am just saying this `if len(podAffinityTerm.TopologyKey) == 0 {` should go away when you removed `if !utilfeature.DefaultFeatureGate.Enabled(features.AffinityInAnnotations)`, as it is already being checked below `if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 {`.